zerver/lib/bugdown/__init__.py

```diff
@@ -1072,7 +1072,11 @@ def run(self, root: Element) -> None:
             # Populate potential_attachment_urls. do_claim_attachments() handles setting has_attachments.
             self.markdown.zulip_message.potential_attachment_urls = []
             for url in unique_urls:
-                self.maybe_append_attachment_url(url)
+                parsed_url = urllib.parse.urlparse(url)
```

Comment:
I think we always want `urlsplit` rather than `urlparse`.

```pycon
>>> urlparse('/a;b')
ParseResult(scheme='', netloc='', path='/a', params='b', query='', fragment='')
>>> urlsplit('/a;b')
SplitResult(scheme='', netloc='', path='/a;b', query='', fragment='')
```